### PR TITLE
remove "everything" tag from tag notifications -- fixes #1692

### LINKED ISSUE #1692: add an exception to list of users to be notified upon new tag addition, for those subscribed to 'everything'
### What happened just before the problem occurred

This is with reference to issue #1630 .
A user subscribed to 'everything' should not get email when a Tag is added to the content because they were already notified when the content was added .

### Relevant URLs

https://github.com/publiclab/plots2/blob/master/app/mailers/subscription_mailer.rb
https://github.com/publiclab/plots2/blob/master/app/models/tag_selection.rb
https://github.com/publiclab/plots2/blob/master/app/models/tag.rb

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
This solves the issue #1692 : add an exception to list of users to be notified upon new tag addition, for those subscribed to 'everything'

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. 
Thanks!

* [X] all tests pass -- `rake test:all`
* [X] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [X] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: app/mailers/subscription_mailer.rb [hunk 1]
```diff
@@ -29,8 +29,11 @@ def notify_tag_added(node, tag, current_user)
     @node = node
     @current_user = current_user
     given_tags = node.tags.reject { |t| t == tag}
-    users_to_email = tag.followers_who_dont_follow_tags(given_tags)
-    users_to_email.each do |user|
+    users_to_email = tag.followers_who_dont_follow_tags(given_tags).pluck(:uid)
```
Comment (jywarren) [inline app/mailers/subscription_mailer.rb]: Is this failing because you can't pluck if it returns empty? Perhaps we have to account for that possibility in this solution.

File: app/models/tag_selection.rb [hunk 2]
```diff
@@ -19,4 +19,10 @@ def ruser
   def tagname
     tag.name
   end
+
+  def self.users_following_everything_tag
```
Comment (jywarren) [inline app/models/tag_selection.rb]: I think you should be able to use `Tag.followers()` instead of a custom method here -- 

https://github.com/sagarpreet-chadha/plots2/blob/6106522b57a4bfe142edb181cf95741d6234ba1e/app/models/tag.rb#L131

File: app/models/tag.rb [hunk 3]
```diff
@@ -215,4 +215,8 @@ def self.trending(limit = 5 , start_date = DateTime.now - 1.month , end_date = D
        .order('count DESC')
        .limit(limit)
   end
+
+  def self.find_id_of_everything_tag
```
Comment (jywarren) [inline app/models/tag.rb]: Also here, I guess I'd just like to suggest using `Tag.find_by_name('everything').tid` instead of making a new one-line model method. I just am not sure we'll be using it again, so why add 3 more lines, you know? I do really like how you're thinking in terms of code modularity though!

**Comment by jywarren** [comment]
Made some suggestions which I hope will help resolve this! Thank you!!!

File: Gemfile.lock [hunk 4]
```diff
@@ -372,4 +372,4 @@ DEPENDENCIES
   will_paginate-bootstrap (>= 1.0.1)
 
 BUNDLED WITH
-   1.14.6
+   1.14.6
```
Comment (jywarren) [inline Gemfile.lock]: Oops, can you drop this file from the PR? Thanks!

File: app/assets/javascripts/application.js [hunk 5]
```diff
@@ -48,4 +48,4 @@
 //= require graph.js
 //= require short-code-forms/dist/short-code-prompts.js
 //= require wikis.js
-//= require chart.js/dist/Chart.js
+//= require chart.js/dist/Chart.js
```
Comment (jywarren) [inline app/assets/javascripts/application.js]: Same here -- i think your editor may be making tweaks, but they don't seem relevant to the PR. Appreciated!

File: app/mailers/subscription_mailer.rb [hunk 6]
```diff
@@ -29,13 +29,24 @@ def notify_tag_added(node, tag, current_user)
...
-    @footer = feature('email-footer')
+
+       final_users_to_email = User.find(final_users_ids) 
+       final_users_to_email.each do |user|
+        @user = user
+          unless user.id == current_user.id 
+             mail(to: user.email, subject: "New tag added on #{node.title}").deliver 
+          end
+        end
+     @footer = feature('email-footer')
```
Comment (jywarren) [inline app/mailers/subscription_mailer.rb]: This all looks good, but if you can be especially careful on line indentations so we keep up a really consistent formatting, that helps readability for future generations of coders! Thanks!

**Comment by jywarren** [comment]
As to the errors above, read through Dangerbot's comments -- see in particular the reference to `users_to_mail` -- and see if you can go through each error and follow what went wrong! 

You're doing great!

File: app/mailers/subscription_mailer.rb (hunk 6 above)
Comment (sagarpreet-chadha) [reply app/mailers/subscription_mailer.rb]: Ok sir...will take care from now onwards ! Thanks !

**Comment by jywarren** [comment]
Moved to https://github.com/publiclab/plots2/pull/1696
